src/bootstrap/test.rs

```diff
@@ -994,7 +1001,23 @@ impl Step for Compiletest {
             cmd.arg("--lldb-python-dir").arg(dir);
         }
 
-        cmd.args(&builder.config.cmd.test_args());
+        // Get paths from cmd args
+        let paths = match &builder.config.cmd {
+            Subcommand::Test { ref paths, ..} => &paths[..],
+            _ => &[]
+        };
+
+        // Get test-args by striping suite path
+        let is_file = |p: &PathBuf| {
+            assert!(p.is_file(), "{:?} is not a valid file", p);
```

Comment:
Let's change the message here to be `Expected {:?} to be a path to a test file` to be clearer about the reason for this error.